addons/website_sale/tests/test_website_sale_gmc.py

```diff
@@ -0,0 +1,434 @@
+        self.blue_sofa.write({
+            'product_variant_image_ids': [
+                Command.create({
+                    'name': f'image {i}',
+                    'image_1920': image,
+                })
+                for i in range(12)
+            ]
+        })
+
+        self.update_items()
+
+        self.assertEqual(10, len(self.blue_sofa_item['additional_image_link']))
+
+    def test_gmc_items_identifier_exists_iff_barcode_exists(self):
+        self.red_sofa.barcode = '0232344532564'
+
+        self.update_items()
+
+        self.assertEqual(self.red_sofa.barcode, self.red_sofa_item['gtin'])
+        self.assertEqual('yes', self.red_sofa_item['identifier_exists'])
+        self.assertNotIn('gtin', self.blue_sofa_item)
+        self.assertEqual('no', self.blue_sofa_item['identifier_exists'])
+
+    def test_gmc_items_sorted_types(self):
+        # Furnitures / Sofas
+        # Furnitures / Indoor Furnitures / Indoor Sofas
+        sofas_categ = self._create_public_category([
+            {'name': 'Furnitures'},
+            {'name': 'Sofas', 'sequence': 1},
+        ])
+        indoor_sofas_categ = self._create_public_category([
+            {'name': 'Furnitures'},
+            {'name': 'Indoor Furnitures'},
+            {'name': 'Indoor Sofas', 'sequence': 2},
+        ])
+        self.public_categories = sofas_categ + indoor_sofas_categ
+        self.product_template_sofa.public_categ_ids = self.public_categories
+
+        self.update_items()
+
+        self.assertListEqual(
+            list(
+                name.replace('/', '>')
+                for name in self.public_categories.sorted('sequence').mapped('display_name')
+            ),
+            self.red_sofa_item['product_type']
+        )
+
+    def test_gmc_items_types_limit_to_5(self):
+        self.product_template_sofa.write({
+            'public_categ_ids': [
+                Command.create({'name': f'Category {i}'}) for i in range(6)
+            ]
+        })
+
+        self.update_items()
+
+        self.assertEqual(5, len(self.red_sofa_item['product_type']))
+
+    def test_gmc_product_variants(self):
+        product_one_variant = self.env['product.template'].create({
+            'name': 'Test product',
+            'attribute_line_ids': [
+                Command.create({
+                    'attribute_id': attr.attribute_id.id,
+                    'value_ids': [Command.link(attr.id)],
+                })
+                for attr in (self.color_attribute_green + self.size_attribute_l)
+            ],
+            'list_price': 49.0,
+        }).product_variant_ids
+        self.products |= product_one_variant
+
+        self.update_items()
+
+        # same template
+        self.assertEqual(self.red_sofa_item['item_group_id'], self.blue_sofa_item['item_group_id'])
+        # no other variant
+        self.assertNotIn('item_group_id', self.items[product_one_variant])
+
+    def test_gmc_items_bundle_iff_combo(self):
```

Comment:
Not a big deal but I don't know how popular "iff" is among non-graduate developers :thinking: